SDL2_OpenGL_BoilerPlate/BoilerPlate/App.cpp

```diff
@@ -66,41 +67,54 @@ namespace Engine
 		}
 
 		// Setup the viewport
-		//
 		SetupViewport();
 
 		// Change game state
-		//
 		m_state = GameState::INIT_SUCCESSFUL;
 
 		return true;
 	}
 
 	void App::OnKeyDown(SDL_KeyboardEvent keyBoardEvent)
-	{		
+	{
 		switch (keyBoardEvent.keysym.scancode)
 		{
 		case SDL_SCANCODE_W:
 			SDL_Log("Up");
-			player->MoveFroward(m_width, m_height);
-			player->setIsThrusterOn(true);
+			game->player->Impulse();
+			game->player->SetIsThrusterOn(true);
 			break;
 
 		case SDL_SCANCODE_A:
 			SDL_Log("Left");
-			player->RotateLeft();
+			game->player->RotateLeft();
 			break;
 
 		case SDL_SCANCODE_D:
 			SDL_Log("Right");
-			player->RotateRight();
+			game->player->RotateRight();
```

Comment:
If you have the `game` object, why do you need to still handle player movement from the App code?